### DISCUSSION TITLE
Welcome to DNAnalyzer Discussions!

### DISCUSSION THREAD

**Discussion by VerisimilitudeX** [VerisimilitudeX/DNAnalyzer#8] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Ast3risk-ops** [comment]
sorry, why isn't this widespread yet?

**Reply by VerisimilitudeX** [reply]
I'm not sure I understand your question completely, but I believe I can answer it. Over the past several weeks, I worked on this algorithm and published it as an open-source project on GitHub. Because it is a new initiative, it has not received as much traffic as other similar ventures. However, based on GitHub's traffic trend, I anticipate hundreds of users in the next month or two. I am constantly looking for new contributors and ideas, so please open an issue or pull request (fork/create a branch first)!
~ Piyush

**Reply by Ast3risk-ops** [reply]
I meant why haven't ppl heard of this this is amazing

also how to use it?

**Reply by VerisimilitudeX** [reply]
> I meant why haven't ppl heard of this this is amazing
> 
> also how to use it?

Ah, I see. Thanks for reaching out!

Do you already have a Java IDE and JDK? If not, I can assist you in quickly setting it up. 

Currently, I'm working on packaging this software with Maven and creating a graphical user interface to make it run like a native app on your computer. Please star/watch this repository if you want to wait until I release it to test it (ETA 1.5 months).

Thanks for reaching out to me again!

**Reply by Ast3risk-ops** [reply]
I do have Java 18 JDK and Java 8 JRE

**Reply by VerisimilitudeX** [reply]
Perfect! Here are the steps to download the software:
1. Download the following OneDrive folder (it's a clone of the GitHub repository): https://1drv.ms/u/s!Ak23YJmC7V17hJhHgcJeb3BIY1CDQg?e=UyyRBi
2. Open the folder `files` in your IDE (VS Code/Intellij Idea/Eclipse)
3. Run the `Main.java` file (if it doesn't work, make sure that you opened the entire folder in your IDE)
4. When prompted for the file name, enter the name of one of the files in the DNA folder (e.g., `brca1line.fa`)
5. When prompted for the amino acid, enter the name of any of the 20 amino acids (e.g., serine, isoleucine, methionine, etc.)
6. Feel free to reach out to me again if you are having any trouble in doing this!

**Reply by Ast3risk-ops** [reply]
Thanks it worked.

**Reply by VerisimilitudeX** [reply]
Excellent. Do you have any ideas/suggestions on how to improve it or did you encounter any bugs? I will be packaging this into a library soon which will allow any user to use the code by just importing it.

**Comment by Ast3risk-ops** [comment]
Hey, so that worked. How would one convert a dna sample (ex. blood) into a string like seen in the .fa files?

**Reply by VerisimilitudeX** [reply]
Excellent question! Here's the process of converting DNA samples into strings:

## Extraction
DNA is stored in the nucleus of all cells. The extraction step is responsible for breaking open the nucleus and releasing the DNA molecules into solution because DNA is found in the nucleus of cells throughout the body. During this step, the DNA molecules is separated from other cellular material and debris that may be present in a biological sample.

## Conversion to strings
Now, coming back to your main question, all 4 nucleotides in DNA (adenine [A], thymine [T], guanine [G], and cytosine [C]) have different shapes, which allows a machine (usually a more complicated type of scanner) to easily differentiate between the letters and convert it to strings. Some other steps were involved in this process, but I omitted them as they aren't as important to understand. However, if you want to read more about it, I would recommend that you check out the following article: https://dps.mn.gov/divisions/bca/bca-divisions/forensic-science/Pages/dna-procedures.aspx

**Reply by Ast3risk-ops** [reply]
Thanks

**Reply by VerisimilitudeX** [reply]
Of course! I would encourage you to check out the new GUI version of this app, which will be available in a few weeks. Perhaps you should star or watch the repository. That version will include genomics lessons and resources like the one above.

**Reply by VerisimilitudeX** [reply]
@Ast3risk-ops we have so many contributors now!

**Reply by VerisimilitudeX** [reply]
@Ast3risk-ops I'd really appreciate it if you could star this repository to support our mission of making open-source DNA analysis tools available to everyone.

**Reply by Ast3risk-ops** [reply]
Done.
